docs/samples/client/csharp/oauth/readme.md

```diff
@@ -0,0 +1,12 @@
+# Guide for CSharp Oath
+
+This guide is intended to assist with creating custom client constructor to support the OAuth token.
+
+## Table of contents
+
+  - [Creating a client with OAuth Token](#oauth-token)
+  - Example implementation of TokenCredential
+  - Example usage of TokenCredential and Custom Client
+
+### Creating a client with OAuth Token
+Link to sample here
```

Comment:
Do we need this separate section here? I think we can just list out the links in the table of contents.

Reply:
Alternatively, we can just inline the code snippet here rather than defining a separate md file for it.